nextcord/interactions/message_component.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Optional, Tuple
+
+from ..errors import InvalidArgument
+from ..message import Message
+from .base import Interaction
+
+__all__ = ("MessageComponentInteraction",)
+
+if TYPE_CHECKING:
+    from ..state import ConnectionState
+    from ..types.interactions import Interaction as InteractionPayload
+
+
+class MessageComponentInteraction(Interaction):
+    """Represents the interaction for all messsage components.
+
+    This interaction is get's triggered by :class:`nextcord.ui.View`
+
+    .. container:: operations
+
+        .. describe:: x == y
+
+            Checks if two interactions are equal.
+
+        .. describe:: x != y
+
+            Checks if two interactions are not equal.
+
+        .. describe:: hash(x)
+
+            Returns the interaction's hash.
+
+    Attributes
+    ----------
+    message: :class:`Message`
+        The message the component is attached to.
+    component_id: :class:`str`
+        The ID of the component that triggered the interaction.
+    value: :class:`str`
+        The value of the component that triggered the interaction.
+    """
+
+    __slots__: Tuple[str, ...] = (
+        "message",
+        "component_id",
+        "value",
+    )
+
+    def __init__(self, *, data: InteractionPayload, state: ConnectionState) -> None:
+        super().__init__(data=data, state=state)
+
+    def _from_data(self, data: InteractionPayload) -> None:
+        super()._from_data(data=data)
+
+        message = data["message"]  # type: ignore # should be present here - not sure why its causing issues
+        self.message = self._state._get_message(
+            int(message["id"])
+        ) or Message(  # ---- check for and create correct message object ----
+            state=self._state, channel=self.channel, data=message  # type: ignore
+        )
+        self.component_id = self.data["custom_id"]  # type: ignore # self.data should be present here
```

Comment:
`self.data` is typed as optional, so according to Pyright it could not be present.